sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -783,6 +783,33 @@ def nsmallest(self, keep, **kwargs):
               preserves_partition_by=partitionings.Arbitrary(),
               requires_partition_by=partitionings.Singleton()))
 
+  @property
+  def is_unique(self):
+    def set_index(s):
+      s = s[:]
+      s.index = s
```

Comment:
Can all types be placed in the index? If not, perhaps we could (sometimes?) set the index to a hash.

Reply:
Thanks thats a good question. I did some local experiments and it looks like anything _can_ be placed in the index. I added some test cases for other dtypes (floats and strings) to be sure.